libs/api/domains/auth/src/lib/meDelegations.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable } from '@nestjs/common'
+
+import { Auth, AuthMiddleware, User } from '@island.is/auth-nest-tools'
+import {
+  DelegationDTO,
+  UpdateDelegationScopeDTO,
+  MeDelegationsControllerFindAllDirectionEnum,
+  MeDelegationsApi,
+} from '@island.is/clients/auth-public-api'
+
+import {
+  UpdateDelegationInput,
+  DeleteDelegationInput,
+  DelegationInput,
+  CreateDelegationInput,
+} from './dto'
+import { DelegationByOtherUserInput } from './dto/delegationByOtherUser.input'
+
+@Injectable()
+export class MeDelegationsService {
+  constructor(private delegationsApi: MeDelegationsApi) {}
+
+  private delegationsApiWithAuth(auth: Auth) {
+    return this.delegationsApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  getDelegations(user: User): Promise<DelegationDTO[]> {
+    return this.delegationsApiWithAuth(user).meDelegationsControllerFindAll({
+      direction: MeDelegationsControllerFindAllDirectionEnum.Outgoing,
+      isValid: true,
+    })
+  }
+
+  getDelegationById(
+    user: User,
+    { delegationId }: DelegationInput,
+  ): Promise<DelegationDTO> {
+    return this.delegationsApiWithAuth(user).meDelegationsControllerFindOne({
+      delegationId,
+    })
+  }
+
+  async getDelegationByOtherUser(
+    user: User,
+    { toNationalId }: DelegationByOtherUserInput,
+  ): Promise<DelegationDTO> {
```

Comment:
This might be undefined, right? I'd suggest this and `getDelegationById` should be consistent, either throw or return null.

Reply:
Fixed